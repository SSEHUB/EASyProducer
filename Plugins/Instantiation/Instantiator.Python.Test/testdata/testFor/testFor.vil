vilScript testFor (Project source, Configuration config, Project target) {

    version v0;

    main(Project source, Configuration config, Project target) = : {   	
    	vilTemplateProcessor("$target/testFor.vtl", config, "$target/testFor.py");
    }
}