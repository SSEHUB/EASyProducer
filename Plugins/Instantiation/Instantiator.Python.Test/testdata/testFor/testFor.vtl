@indent(indentation = 4, additional = 0)
@format(profile="python")
template testFor(Configuration config, PythonCodeArtifact target) {
    def main(Configuration config, PythonCodeArtifact target) {
    	target.addAssign("arr", "[0,1,2,3,4,5]");
    	target.addFor("elem", "arr")
    		.addCall("print").addArgument("elem");

    	target.addFor("elem", "0", "4")
    		.addCall("print").addArgument("elem");

    	target.addFor("elem", "0", "6", "2")
    		.addCall("print").addArgument("elem");
    }
}
