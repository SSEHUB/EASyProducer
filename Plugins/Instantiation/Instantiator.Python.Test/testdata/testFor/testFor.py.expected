arr = [0,1,2,3,4,5]
for elem in arr:
    print(elem)
for elem in range(0, 4):
    print(elem)
for elem in range(0, 6, 2):
    print(elem)
